### PR TITLE
[test] Update hdf5 test to include prgenv nvidia

### PR DESCRIPTION
**Pull request by rsarm**

### CODE REVIEW

File: cscs-checks/libraries/io/hdf5_compile_run.py [hunk 1]
```diff
@@ -7,35 +7,47 @@
...
+    tags = {'production', 'craype', 'health'}
 
+    @run_after('init')
+    def add_valid_systems(self):
+        self.descr = (self.lang_names[self.lang] + ' HDF5 ' +
+                      self.linkage.capitalize())
+        self.sourcepath = f'h5ex_d_chunk.{self.lang}'
+        if self.linkage == 'dynamic':
+            self.valid_systems += ['eiger:mc', 'pilatus:mc']
+        self.build_system.ldflags = [f'-{self.linkage}']
```
Comment (ekouts) [inline cscs-checks/libraries/io/hdf5_compile_run.py:36]:
```suggestion
            self.valid_systems += ['eiger:mc', 'pilatus:mc']

        self.build_system.ldflags = [f'-{self.linkage}']
```

**Review by ekouts — APPROVED** [review]
lgtm

**Comment by teojgo** [comment]
@rsarm I have updated that on #2024

**Comment by jjotero** [comment]
This check was already updated in #2024

**Pull request closed without merging**